src/firefly/js/templates/lightcurve/LcViewer.jsx

```diff
@@ -250,13 +251,23 @@ UploadPanel.defaultProps = {
     name: 'LCUpload'
 };
 
+export function getUploadedFileName(){
+    var fields = FieldGroupUtils.getGroupFields(vFileKey);
+    const displayName = fields.rawTblSource.displayValue.split('\\');
+    return ( displayName && displayName[displayName.length-1])?displayName[displayName.length-1]:'';
```

Comment:
Using the value from input field as if it's static is not accurate.  The value can be changed.
Try select file 'a', then upload.  Then select file 'b', but click cancel.  Then, click on 'Period Finder' -> cancel, then you will see the uploaded file name is now 'b' when it should have been 'a'.